### PR TITLE
Moving uptime to PoS

### LINKED ISSUE #496: refactor staking info out of ValidatorManager
**Context and scope**
Follow up from https://github.com/ava-labs/teleporter/pull/491, which starts the initial inheritance hierarchy between the contracts. `ValidatorManager` should only keep state/logic that applies to both PoA and PoS validator managers, in order to support migration from PoA -> PoS #488 related. 

At least the `uptime` and `rewarded` infos on the `ValidatorInfo` is currently saved in the base contract's state, and only applies to PoS. 

**Discussion and alternatives**
Could have separate state variables that keep track of all stake related info for a validation ID. Will likely live in `PoSValidatorManager` as it does not apply to PoA, and applies to both native or erc20 token staking.

### PR DESCRIPTION
**Pull request by minghinmatthewlam**
## Why this should be merged
Fixes #496 

## How this works
Moves PoS logic out of `ValidatorManager` for uptime. Validator churn tracking will be moved in #480 

Verifies the uptime proof and sets the uptime if valid, then calls the underlying initialize end validation.

## How this was tested
Unit tests for uptime

## How is this documented

### CODE REVIEW

File: contracts/staking/PoSValidatorManager.sol [hunk 1]
```diff
@@ -21,6 +21,7 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         uint256 _maximumStakeAmount;
         uint64 _minimumStakeDuration;
         IRewardCalculator _rewardCalculator;
+        mapping(bytes32 validationID => uint64) _uptimes;
```
Comment (bernard-avalabs) [inline contracts/staking/PoSValidatorManager.sol:24]: small nit: perhaps validatorUptimes would be more descriptive.

**Review by bernard-avalabs — COMMENTED** [review]
Generally makes sense to me. Looks like there are a few Slither alerts to cleanup.

**Review by geoff-vball — APPROVED** [review]
Generally LGTM 👍

File: contracts/staking/ValidatorMessages.sol [hunk 2]
```diff
@@ -7,8 +7,8 @@ pragma solidity 0.8.25;
 library ValidatorMessages {
     // The information that uniquely identifies a subnet validation period.
     // The validationID is the SHA-256 hash of the concatenation of the CODEC_ID,
-    // REGISTER_SUBNET_VALIDATOR_MESSAGE_TYPE_ID, and the concatenated ValidationInfo fields.
-    struct ValidationInfo {
+    // REGISTER_SUBNET_VALIDATOR_MESSAGE_TYPE_ID, and the concatenated ValidationPeriod fields.
+    struct ValidationPeriod {
```
Comment (cam-schultz) [inline contracts/staking/ValidatorMessages.sol:11]: I like this name. We should adopt this nomenclature whenever we're describing the entity specified by `validationID`.

**Review by cam-schultz — APPROVED** [review]

**Pull request merged**